### PR TITLE
Major rework

### PR DESCRIPTION
**Pull request by dafortin**
#major#

# CHANGELOG
* Add a few folder to `.gitignore`
* Fix issue #9
	* Remove `ember-frost-theme` and update `ember-frost-core` in `package.json`
	* Import css from core
* Run tests on Chrome in CI
* Add `CONTRIBUTING.md` and `PULL_REQUEST_TEMPLATE.md` 
* Remove unnecessary route and controller
* Clean integration tests
* Increase dependencies version
	* `ember-cli`
	* `liquid-fire`
* Switch `from blanket` to `ember-cli-code-coverage`
* Change baseURL to rootURL
* User `ember-hook` in integration tests
* Change interface to take a component as the content
* Add following action hooks:
	* onClose
	* onOpen
	* onToggle
* Rework style
* Resizing based on content
* Linting
	* Add sass-linting
	* Add eslint
	* Add template linting
	* Fix linting issues
* Add visual acceptance and modify tests
* Update README
* Add ember hooks
* Reword demo application
* Add more tests

### CODE REVIEW

File: addon/styles/_frost-sidebar.scss [hunk 1]
```diff
@@ -1,74 +1,69 @@
 .frost-sidebar {
   position: absolute;
-  left: 0px;
+  max-width: 350px;
+  min-height: 100px;
+  max-height: 250px;
```
Comment (sglanzer) [inline addon/styles/_frost-sidebar.scss]: Likely we'll override these at some point, might be worth documenting in the interface

File: addon/styles/_frost-sidebar.scss [hunk 2]
```diff
@@ -1,74 +1,69 @@
...
-      .ps-scrollbar-x-rail{
-        background-color: $frost-color-white;
-        position:relative;
-      }
-
-      .ps-scrollbar-y-rail{
-        background-color: $frost-color-white;
-        position:absolute;
-      }
+      max-width: 300px;
```
Comment (sglanzer) [inline addon/styles/_frost-sidebar.scss]: Can these be set as 100% and have the container be `position: relative`?

File: addon/templates/components/frost-sidebar.hbs [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
-        <div class="pod-content">
-          {{yield}}
+        <div class="frost-sidebar-content" data-test={{hook (concat hook '-sidebar-content')}}>
+          {{component content}}
         </div>
       {{/frost-scroll}}
     </div>
-    <div class="tab tab-open" {{action (action 'toggleSidebar')}}>
-      {{frost-icon class="icon" icon="frost/sidebar-layer-open"}}
+    <div class="frost-sidebar-tab frost-sidebar-tab-open" {{action onToggle}} data-test={{hook (concat hook '-sidebar-tab')}}>
```
Comment (sglanzer) [inline addon/templates/components/frost-sidebar.hbs]: The `tab` is probably more accurately described as a `button`, but it's up to you if you feel inspired to change that

File: tests/dummy/app/index.html [hunk 4]
```diff
@@ -3,22 +3,22 @@
...
-    <title>frost-sidebar testbed</title>
+    <title>Frost Sidebar Demo</title>
     <meta name="description" content="">
     <meta name="viewport" content="width=device-width, initial-scale=1">
 
     {{content-for 'head'}}
 
-    <link rel="stylesheet" href="assets/vendor.css">
-    <link rel="stylesheet" href="assets/dummy.css">
+    <link rel="stylesheet" href="{{rootURL}}assets/vendor.css">
```
Comment (sglanzer) [inline tests/dummy/app/index.html]: I would have thought these needed a `/`, but I guess not

File: bower.json [hunk 5]
```diff
@@ -1,22 +1,21 @@
 {
   "name": "ember-frost-sidebar",
   "dependencies": {
-    "ember": "~2.3.0",
+    "ember": "~2.7.0",
```
Comment (sglanzer) [inline bower.json]: Ah, we also need this to go to 2.8

File: tests/dummy/app/index.html (hunk 4 above)
Comment (dafortin) [inline tests/dummy/app/index.html]: No we don't need the `/` since rootURL is defined as `rootURL: '/',`

File: bower.json (hunk 5 above)
Comment (dafortin) [inline bower.json]: I tried to increase this to 2.8, but I'm getting a lot of issues with the liquid-fire error. I think I will raise a bug for now to increase Ember to 2.8 and fix this liquid fire error. Will that be ok?

File: addon/templates/components/frost-sidebar.hbs (hunk 3 above)
Comment (dafortin) [inline addon/templates/components/frost-sidebar.hbs]: Yes, always feel like it to do this kind of improvements :) Clarity is my best friend in code :P

File: bower.json (hunk 5 above)
Comment (sglanzer) [inline bower.json]: Yeah, that's fine, I just didn't want to lose track of it

File: addon/templates/components/frost-sidebar.hbs (hunk 3 above)
Comment (sglanzer) [inline addon/templates/components/frost-sidebar.hbs]: I find proper naming helps a **ton** when dealing with CSS

File: bower.json (hunk 5 above)
Comment (dafortin) [inline bower.json]: I will make sure to document that as a bug and as clear as possible.

File: addon/templates/components/frost-sidebar.hbs (hunk 3 above)
Comment (dafortin) [inline addon/templates/components/frost-sidebar.hbs]: Agree !

File: addon/styles/_frost-sidebar.scss (hunk 1 above)
Comment (dafortin) [inline addon/styles/_frost-sidebar.scss]: Done

File: bower.json (hunk 5 above)
Comment (dafortin) [inline bower.json]: Issue #16 raised

**Comment by sglanzer** [comment]
Approved - thanks!

**Pull request merged**